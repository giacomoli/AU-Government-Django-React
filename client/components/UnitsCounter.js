import React from 'react'
import { graphql, gql } from 'react-apollo'

import withData from 'lib/withData'
import withAuth from 'lib/withAuth'
import { userFragment } from 'fragments'
import LineLoader from 'components/LineLoader'

const getUser = gql`
  query me {
    me {
      ...User
    }
  }
${userFragment}
`

class UnitsCounter extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      user: null,
    }
  }
  render() {
    if (this.props.userData.loading) {
      return (<LineLoader />)
    }
    return (
      <div className="row mt-2 mb-3">
        <div className="col text-center">
          User Units Today: {this.props.userData.me.unitsToday}
        </div>
        <div
          className="col text-center"
          style={{ marginLeft: 'auto', borderLeft: 'solid 1px #adb0c6', borderRight: 'solid 1px #adb0c6' }}
        >
          User Units this week: {this.props.userData.me.unitsWeek}
        </div>
        <div className="col text-center" style={{ marginLeft: 'auto' }}>
          User Units this month: {this.props.userData.me.unitsMonth}
        </div>
      </div>
    )
  }
}

export default withData(
  withAuth(
    graphql(getUser, { name: 'userData', options: { fetchPolicy: 'network-only', skip: false } })(
      UnitsCounter,
    ),
  ),
)
