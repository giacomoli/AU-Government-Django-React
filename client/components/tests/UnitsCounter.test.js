import React from 'react'
import renderer from 'react-test-renderer'
import UnitsCounter from '../UnitsCounter'

/* eslint-disable */
describe('UnitsCounter', () => {

  it('should render UnitsCounter with props', () => {
    const props = {
      unitsToday: 10,
      unitsWeek: 11,
      unitsMonth: 12,
    }
    const wrapper = renderer.create(
        <UnitsCounter
          user={props}
        />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
