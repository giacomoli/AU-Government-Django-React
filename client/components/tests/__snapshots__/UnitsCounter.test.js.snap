// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnitsCounter should render UnitsCounter with props 1`] = `
<div
  className="preloader"
>
  <svg
    className="circular"
    viewBox="25 25 50 50"
  >
    <circle
      className="path"
      cx="50"
      cy="50"
      fill="none"
      r="20"
      strokeMiterlimit="10"
      strokeWidth="2"
    />
  </svg>
</div>
`;
